### PR TITLE
Doc for Az framework

### PR DESCRIPTION
**Pull request by bradolenick**

### CODE REVIEW

File: portal-sdk/templates/top-az-framework.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+* With '`Az`', there has been a focus to reduce FX-attributable perf cost to an absolute minimum, putting partner teams in exclusive control over their Blade/Part rendering perf.
+* Only where absolutely necessary (for perf, for security) does '`Az`' introduce Ibiza-specific steps beyond conventional, modern web UI development, imposing minimal constraints on how you choose to develop your web UI.
+
+Here, you develop your Blade or Part as a standard webpage that the Azure Portal will load in a child IFrame.  You pick your favorite OSS/FX support.  You choose your own hosting solution (and you can use Ibiza's hosting service, if you'd like).
+
+From a 10,000-foot view, there is a single additional step you take when developing your Blade or Part UI as a standard HTML webpage.  With this solution, we require that you load a single '`Az`' script into your webpage and call the '`Az.initialized()`' API to signal to the Azure Portal that your rendering is complete.
+
+{"gitdown": "include-section", "file": "../Samples/SampleAzExtension/Src/Components/SampleBlade.tsx", "section": "docs#RequireAz"}
```
Comment (ShrishaKumar) [inline portal-sdk/templates/top-az-framework.md]: You will have to add a symbolic link for the snippet injection to work here.

File: portal-sdk/templates/top-az-framework.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+    // ...
+
+{"gitdown": "include-section", "file": "../Samples/SampleAzExtension/Src/Components/SampleBlade.tsx", "section": "docs#CallAzInitialized"}
+
+This '`Az`' script is a new, [UMD](https://github.com/umdjs/umd) module that provides a minimal **hosting API** for your Blade and Part UI.  You'll load this new library into your HTML page from a URL passed as part of the IFrame '`src`' URL.  You'll obtain TypeScript type definitions for the new '`Az`' API via the '`@types/Az`' NPM package (TODO: Add links throughout to NPM packages).
+
+The API itself resembles the 'container' API exposed to classic Blades and Parts and includes: authentication tokens, theming, Blade-opening/navigation, CommandBar, Dialogs, etc.  You can explore the breadth of these APIs in [this sample page](https://df.onecloud.azure-test.net/#blade/SampleAzExtension/SampleBlade).
```
Comment (ShrishaKumar) [inline portal-sdk/templates/top-az-framework.md]: This link does not work.
image

File: portal-sdk/templates/top-az-framework.md (hunk 1 above)
Comment (bradolenick) [reply portal-sdk/templates/top-az-framework.md]: Done.

File: portal-sdk/templates/top-az-framework.md (hunk 2 above)
Comment (bradolenick) [reply portal-sdk/templates/top-az-framework.md]: It will on the next dogfood deployment.

File: portal-sdk/templates/top-az-framework.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+
+Here, you develop your Blade or Part as a standard webpage that the Azure Portal will load in a child IFrame.  You pick your favorite OSS/FX support.  You choose your own hosting solution (and you can use Ibiza's hosting service, if you'd like).
+
+From a 10,000-foot view, there is a single additional step you take when developing your Blade or Part UI as a standard HTML webpage.  With this solution, we require that you load a single '`Az`' script into your webpage and call the '`Az.initialized()`' API to signal to the Azure Portal that your rendering is complete.
+
+{"gitdown": "include-section", "file": "../Samples/SampleAzExtension/Src/Components/SampleBlade.tsx", "section": "docs#RequireAz"}
+
+    // ...
+
+{"gitdown": "include-section", "file": "../Samples/SampleAzExtension/Src/Components/SampleBlade.tsx", "section": "docs#CallAzInitialized"}
```
Comment (nickharris) [inline portal-sdk/templates/top-az-framework.md]: add tsx to snippet injection map for gitdown

Comment (bradolenick) [reply portal-sdk/templates/top-az-framework.md]: Great catch!

File: portal-sdk/templates/top-az-framework.md [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+* '`true`' (default if '`clientOptimizations`' is not supplied) - Scripts should be bundled and minified
+* '`false`' - Scripts should not be bundled or minified
+* '`bundle`' - Scripts should only be bundled but not minified
+
+In scenario #2, where Blades and Parts are typically hosted by Ibiza's hosting service, this treatment of '`clientOptimizations`' is built-in.  In scenario #1, you will have to integrate '`clientOptimizations`' with your site's bundling/minification abilities.
+
+# Wrap-up
+
+Feedback always welcome.
+Post to [internal StackOverflow](https://stackoverflow.microsoft.com/questions/tagged/ibiza-az) with questions and difficulties.
```
Comment (adamabmsft) [inline portal-sdk/templates/top-az-framework.md]: We should update the officially supported tags doc to include this SO tag.

Comment (bradolenick) [reply portal-sdk/templates/top-az-framework.md]: Done

**Pull request merged**